### DISCUSSION TITLE
Weird interaction with -J (--remote-header-name)

### DISCUSSION THREAD

**Discussion by Kandreas9** [curl/curl#14044] [General]
Most product download buttons contain a redirect to the actual file download url, ex:

`https://example.com/file`

Trying to use curl to get the file with such a link, will cause the downloaded file to be named after the slug, which in this example would download an extensionless file called `file`.

Looking up for solutions I found the -J flag, which says that it uses the content-disposition header and names it accordingly.

The issue I am having, or at least what I believe the issue might be. Is the case where the header doesn't exist causing the file name to default to the slug.

My real world example is with the iTerm2 download link:

`https://iterm2.com/downloads/stable/latest`

This is the link taken directly from their download button, which redirects to this link:

`https://iterm2.com/downloads/stable/iTerm2-3_5_2.zip`

Result:
A file called 
`latest`

Expected result:
A file called
`iTerm2-3_5_2.zip`

**Comment by dfandrich** [comment]
https://curl.se/docs/todo.html#consider_filename_from_the_redir
